Feature/start timer

eniluck opened the pull request.

eniluck committed:
наброски

eniluck committed:
копилируется

eniluck committed:
feat: добавил историю для задач

eniluck committed:
feat: Первая реализация старта таймера

pingvin1308 commented on Pomodoro.Api/Controllers/TimerController.cs:
```diff
@@ -0,0 +1,35 @@
+    [ApiController]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [Produces(MediaTypeNames.Application.Json)]
+    public class TimerController : ControllerBase
```
PomodoroController

pingvin1308 commented on Pomodoro.Api/Controllers/TimerController.cs:
```diff
@@ -0,0 +1,35 @@
+        }
+
+        [HttpGet("{taskId:int}")]
+        public async Task<IActionResult> Start(int taskId)
```
HttpPost("{taskId:int}")

pingvin1308 commented on Pomodoro.Api/ExceptionHandlingFilter.cs:
```diff
@@ -5,10 +5,16 @@ internal class ExceptionHandlingFilter : IExceptionFilter
+        var message = context.Exception.Message;
+        if (context.Exception.InnerException != null)
+        {
+            message += $" Inner exception: {context.Exception.InnerException.Message}";
```
Проверить ToString() для Exception. По моему он должен сам развернуть все InnerExceptions

eniluck replied:
Так и есть.

pingvin1308 commented on Pomodoro.Core/ITimerService.cs:
```diff
@@ -0,0 +1,6 @@
+
+public interface ITimerService
+{
+    public Task StartAsync(int taskId);
```
public можно убрать

pingvin1308 commented on Pomodoro.DAL.Postgres/Configurations/TaskHistoryConfiguration.cs:
```diff
@@ -0,0 +1,34 @@
+            .WithMany();
+
+        builder.Property(x => x.CreateDateTime)
+            .HasColumnType("timestamp with time zone")
```
timestamp with time zone -> timestamp without time zone

pingvin1308 commented on Pomodoro.DAL.Postgres/PomodoroDbContext.cs:
```diff
@@ -14,12 +14,15 @@ public PomodoroDbContext(DbContextOptions<PomodoroDbContext> options)
         {
             base.OnModelCreating(modelBuilder);
 
             modelBuilder.ApplyConfiguration(new TaskCategoryConfiguration());
```
modelBuilder.ApplyConfigurationFromAssebly()

eniluck committed:
Update dotnet.yml

Update dotnet.yml

Update dotnet.yml

Update dotnet.yml

Update dotnet.yml

Update dotnet.yml

eniluck committed:
feat: много изменений

- доработал логику старта помидорки
- подключил CSharpFunctionalExtensions
- utc в без timezone в бд.
- исправил обновление taskEntity перед добавлением истории
- удалил поле
- вывел общий класс для контроллера

eniluck committed:
feat: метод start контроллера Pomodoro теперь POST

eniluck committed:
feat: Все конфигурации для сущностей в PomodoroDbContext применяются из сборки.

eniluck committed:
fix: небольшое изменение
